event_routing_backends/helpers.py

```diff
@@ -48,20 +75,28 @@ def get_anonymous_user_id(username_or_id, external_type):
     elif user_id:
         user = User.objects.filter(id=user_id).first()
 
-    if not user:
-        logger.info('User with username "%s" does not exist. '
-                    'Cannot generate anonymous ID', username_or_id)
+    return user
 
-        anonymous_id = str(uuid4())
-    else:
-        type_name = getattr(ExternalIdType, external_type)
-        external_id, _ = ExternalId.add_new_user_id(user, type_name)
-        if not external_id:
-            raise ValueError("External ID type: %s does not exist" % type_name)
 
-        anonymous_id = str(external_id.external_user_id)
+def get_user_email(username_or_id):
+    """
+    Get user's email from username or user id.
 
-    return anonymous_id
+    Arguments:
+        username_or_id (str):     username for the learner
+
+    Returns:
+        str
+    """
+    user = get_user(username_or_id)
+
+    if not user:
+        logger.info('User with username "%s" does not exist.', username_or_id)
```

Comment:
I'm slightly concerned about this case. I can see it happening in the event of log replay with retired users, who get their username changed, but can you think of any other case where it may happen? If it's only retired users, we may be able to change the username case in get_user to use this to make the user consistent: https://github.com/openedx/edx-platform/blob/bf36c429509b92854799d5d9b5203b4eb9d7e0df/common/djangoapps/student/models/user.py#L317

Reply:
@bmtcril I can't think of anyother use case. I have a added a check to get retired user if user is not found in normal flow.